src/main/java/sirius/pasta/noodle/compiler/TypeTools.java

```diff
@@ -114,7 +114,13 @@ private void propagateConstantValueTypeInfos(Type parameterType, Node parameter)
 
         String typeVariableName = ((TypeVariable<?>) parameterType).getName();
         Type genericType = parameter.getGenericType();
-        typeTable.put(typeVariableName, genericType == null ? parameter.getType() : genericType);
+        Type type = genericType == null ? parameter.getType() : genericType;
+
+        if ("void".equals(type.getTypeName())) {
```

Comment:
wouldn't work `type != void.class` here? And in this case we can skip the putIfAbsent entirely as this doesn't narrow down anything, right?